python/sanlock.c

```diff
@@ -1395,15 +1438,20 @@ py_killpath(PyObject *self __unused, PyObject *args, PyObject *keywds)
 
     /* configure killpath (gil disabled) */
     Py_BEGIN_ALLOW_THREADS
-    rv = sanlock_killpath(sanlockfd, 0, path, kpargs);
+    rv = sanlock_killpath(sanlockfd, 0, PyBytes_AsString(path), kpargs);
     Py_END_ALLOW_THREADS
+    Py_DECREF(path);
 
     if (rv != 0) {
         __set_exception(rv, "Killpath script not configured");
         return NULL;
     }
 
     Py_RETURN_NONE;
+
+exit_fail:
+    Py_XDECREF(path);
+    return NULL;
```

Comment:
We can use rv here to set the exeption and return the correct value:

    out:
        Py_DECREF(path)  // cannot be null
        if (rv != 0) {
            fail...
        }

        Py_RETURN_NONE